include/beast/websocket/impl/ping.ipp

```diff
@@ -146,24 +146,31 @@ operator()(error_code ec, bool again)
                         boost::asio::error::operation_aborted));
                 return;
             }
-            // fall through
+            d.ws.wr_block_ = &d;
+            // [[fallthrough]]
 
         case 1:
             // send ping frame
             d.state = 99;
-            BOOST_ASSERT(! d.ws.wr_block_);
-            d.ws.wr_block_ = &d;
             boost::asio::async_write(d.ws.stream_,
                 d.fb.data(), std::move(*this));
             return;
 
         case 2:
+            BOOST_ASSERT(! d.ws.wr_block_);
```

Comment:
I'm lost. I see that `wr_block_` is cleared in the `upcall`. But when I look on line 132 of this commit, I see that if we have a `wr_block_`, then we set the state to `2`. The time time into this state machine I would expect this assert to fail. What I'm missing is how is `wr_block_` cleared between state `0` and state `2`?